src/analysis/processing/qgsalgorithmsplitvectorlayer.cpp

```diff
@@ -139,13 +139,15 @@ QVariantMap QgsSplitVectorLayerAlgorithm::processAlgorithm( const QVariantMap &p
     }
     else
     {
-      fileName = QStringLiteral( "%1%2.%3" ).arg( baseName ).arg( ( *it ).toString() ).arg( outputFormat );
+      QString value = ( *it ).toString();
+      value.replace( QRegularExpression( "<|>|\\:|\\\\|\\/|\\||\\?|\\*|\"" ), QStringLiteral( "_" ) );
```

Comment:
Can we use QgsFileUtils::stringToSafeFilename here instead?

Reply:
Sure, I have updated code.